### PR TITLE
Ga 107 profile refresh queue repo

### PR DESCRIPTION
**Pull request by abul-syed**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/GA-107

### Change description ###

PRM : Write to profile_refresh_queue table a record for each profile ID identified

### CODE REVIEW

**Review by patelila — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        "classpath:sql/insert_organisation_profiles.sql"
+    })
+    public void shouldHandleDuplicateOrganisationProfileIds() {
+        List<ProfileRefreshQueueEntity> profileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(profileRefreshQueueEntities, 1, 1);
+
+        List<String> organisationProfileIds = List.of("SOLICITOR_ORG");
+        String orgProfileIds = String.join(",", organisationProfileIds);
+
+        profileRefreshQueueRepository.upsertOrganisationProfileIds(orgProfileIds, 2);
```
Comment (sanjayparekh-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java:40]: is it worth adding a test for when you dont supply the correct upsert version?

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        List<ProfileRefreshQueueEntity> newProfileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(newProfileRefreshQueueEntities, 1, 2);
+    }
+
+    private void validateData(List<ProfileRefreshQueueEntity> profileRefreshQueueEntities,
+                                int expectedSize,
+                                int expectedAccessTypesMinVersion) {
+        assertEquals(expectedSize, profileRefreshQueueEntities.size());
+        profileRefreshQueueEntities.forEach(el -> {
```
Comment (helenbirdhmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java:50]: rename variable el

Comment (abul-syed) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java]: renamed in commit [c2f979a](https://github.com/hmcts/am-org-role-mapping-service/pull/1729/commits/c2f979a5eab2a5c11e7ea4ca33518d905a52e734)

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java (hunk 1 above)
Comment (abul-syed) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java:40]: the version that’s used in the upsert query will comes from the below query:

```
update access_types
set version = version + 1, access_types = ?::jsonb
returning version, access_types::text
```

so only when we update the above table, we then use the returned version value. but this table can only be updated once at a time due to the database lock in the below query (`for update`)

```
select version as version, access_types::text as
access_types
from access_types
for update
```

due to the database locking, the first update query should return a consistent and increasing upsert version

**Review by helenbirdhmcts — APPROVED** [review]
lgtm

**Review by sanjayparekh-hmcts — APPROVED** [review]

**Comment by TomForniCGI** [comment]
Merged into GA-54

**Pull request closed without merging**